Detectors/HMPID/calibration/testWorkflow/HMPIDDCSDataProcessorSpec.h

```diff
@@ -0,0 +1,304 @@
+      mProcessor->clearCCDBObjects(); // clears the vectors
+
+      // ef : should the unordered map of DPs also be cleared?
+      // (should init() in this class then be called?)
+    }
+  }
+
+  //==========================================================================
+
+  void endOfStream(o2::framework::EndOfStreamContext& ec) final
+  {
+    mProcessor->finalize();
+
+    sendChargeThresOutput(ec.outputs());
+    sendRefIndexOutput(ec.outputs());
+  }
+
+  //==========================================================================
+
+ private:
+  // fill CCDB with ChargeThresh (arQthre)
+  void sendChargeThresOutput(o2::framework::DataAllocator& output)
+  {
+    const auto& payload = mProcessor->getChargeCutObj();
+    auto& info = mProcessor->getHmpidChargeInfo();
+
+    auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
+    LOG(info) << "Sending object " << info.getPath() << "/"
+              << info.getFileName() << " of size " << image->size()
+              << " bytes, valid for " << info.getStartValidityTimestamp()
+              << " : " << info.getEndValidityTimestamp();
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBPayload, "ChargeCut", 0},
+      *image.get());
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBWrapper, "ChargeCut", 0},
+      info);
+  }
+
+  // fill CCDB with RefIndex (arrMean)
+  void sendRefIndexOutput(o2::framework::DataAllocator& output)
+  {
+    // fill CCDB with RefIndex (std::vector<TF1> arrMean)
+    const auto& payload = mProcessor->getRefIndexObj();
+    auto& info = mProcessor->getccdbRefInfo();
+
+    auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
+    LOG(info) << "Sending object " << info.getPath() << "/"
+              << info.getFileName() << " of size " << image->size()
+              << " bytes, valid for " << info.getStartValidityTimestamp()
+              << " : " << info.getEndValidityTimestamp();
+
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBPayload, "RefIndex", 0},
+      *image.get());
+    output.snapshot(
+      Output{o2::calibration::Utils::gDataOriginCDBWrapper, "RefIndex", 0},
+      info);
+  }
+
+  std::vector<std::string> aliases = {
+    "HMP_ENV_PENV",
+    "HMP_MP[0..6]_GAS_PMWPC",
+    "HMP_MP[0..6]_LIQ_LOOP_RAD_[0..2]_IN_TEMP",
+    "HMP_MP[0..6]_LIQ_LOOP_RAD_[0..2]_OUT_TEMP",
+    "HMP_MP_[0..6]_SEC_[0..5]_HV_VMON",
+    "HMP_TRANPLANT_MEASURE_[0..29]_WAVELENGHT",
+    "HMP_TRANPLANT_MEASURE_[0..29]_ARGONREFERENCE",
+    "HMP_TRANPLANT_MEASURE_[0..29]_ARGONCELL",
+    "HMP_TRANPLANT_MEASURE_[0..29]_C6F14REFERENCE",
+    "HMP_TRANPLANT_MEASURE_[0..29]_C6F14CELL"
+    "HMP_RUNNUMBER"};
+
+  bool isRunStarted = false;
+  bool mCheckRunStartStop = false;
+  o2::dcs::RunStatusChecker mRunChecker{o2::detectors::DetID::getMask("HMP")};
+
+  std::unique_ptr<HMPIDDCSProcessor> mProcessor;
+  HighResClock::time_point mTimer;
+  int64_t mDPsUpdateInterval;
```

Comment:
I don't see where you set or use update interval. Please avoid very frequent updates of CCDB at every change. See how other detectors, e.g. FIT/FT0/calibration/testWorkflow/FT0DCSDataProcessorSpec.h do

Reply:
Hi @shahor02, variable mDPsUpdateInterval is not used indeed, is there by mistake, we are going to remove it! We update CCDB just at the end of RUN (see endOfStream method). I think this is correct! At least if I have not misunderstood something!

Reply:
Hi @gvolpe79 
The endOfStream for the DP processor is rather meaningless: the DP processing workflow runs continuously and is not synchronized with SOR/EOR. The only way to detect run start/stop is to use `mRunChecker.getRunStatus()` as you do and send run-related output from the run(..) method. That said, note that it is not 100% reliable: if the AliECS crashes, the EOR will be never flagged in the CCDB. @knopers8 works on fixing this but the ETA is not clear.